sdk/nodejs/resource.ts

```diff
@@ -987,6 +994,60 @@ export interface CustomResourceOptions extends ResourceOptions {
     // that mergeOptions works properly for it.
 }
 
+export class ResourceHook {
+    public name: string;
+    public handler: ResourceHookFunction;
+    public onDryRun: boolean;
```

Comment:
We should doc comment public classes, properties, methods, etc. throughout if we can.

Reply:
Yes, I'll copy the ones from Python once we're happy with that PR.